scaleway/resource_mnq_sns_topic_subscription.go

```diff
@@ -0,0 +1,213 @@
+package scaleway
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/service/sns"
+	"github.com/hashicorp/go-cty/cty"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	mnq "github.com/scaleway/scaleway-sdk-go/api/mnq/v1beta1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+func resourceScalewayMNQSNSTopicSubscription() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceScalewayMNQSNSTopicSubscriptionCreate,
+		ReadContext:   resourceScalewayMNQSNSTopicSubscriptionRead,
+		DeleteContext: resourceScalewayMNQSNSTopicSubscriptionDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"protocol": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Protocol of the SNS Topic Subscription.", // TODO: add argument list
+				ForceNew:    true,
+			},
+			"endpoint": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Endpoint of the subscription",
+				ForceNew:    true,
+			},
+			"sns_endpoint": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "https://sns.mnq.{region}.scaleway.com",
+				Description: "SNS endpoint",
+				ForceNew:    true,
+			},
+			"topic_arn": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				AtLeastOneOf: []string{"topic_id"},
+				ForceNew:     true,
+			},
+			"topic_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				AtLeastOneOf: []string{"topic_arn"},
+				ForceNew:     true,
+			},
+			"access_key": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Sensitive:   true,
+				Description: "SNS access key",
+				ForceNew:    true,
+			},
+			"secret_key": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Sensitive:   true,
+				Description: "SNS secret key",
+				ForceNew:    true,
+			},
+			"redrive_policy": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Optional:    true,
+				Description: "JSON Redrive policy",
+				ForceNew:    true,
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
```

Comment:
Could you add a description here? In particular with the format that will be returned